snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
```

Comment:
ok. And if you need the sideeffect for efficiency, maybe split it in two: one _get, and one _update that calls the _get. Or if it makes a bigger mess, just add a docstring to what you have explaining the side effect.